sdk/core/azure-core/test/ut/base64_test.cpp

```diff
@@ -125,3 +125,41 @@ TEST(Base64, Roundtrip)
     EXPECT_EQ(Convert::Base64Decode(Convert::Base64Encode(data)), data);
   }
 }
+
+TEST(Base64, InvalidDecode)
+{
+  EXPECT_NO_THROW(Convert::Base64Decode(Convert::Base64Encode(std::vector<uint8_t>{})));
+  EXPECT_NO_THROW(Convert::Base64Decode(""));
+  EXPECT_THROW(Convert::Base64Decode("a"), std::runtime_error);
+  EXPECT_THROW(Convert::Base64Decode("aa"), std::runtime_error);
+  EXPECT_THROW(Convert::Base64Decode("aaa"), std::runtime_error);
+  EXPECT_THROW(Convert::Base64Decode("a==="), std::runtime_error);
+  EXPECT_NO_THROW(Convert::Base64Decode("aa=="));
+  EXPECT_NO_THROW(Convert::Base64Decode("aaa="));
```

Comment:
Same here, consider moving these out to a different test, doesn't fit well with `InvalidDecode` unit test.